tests/unit/packet/PublishTest.php

```diff
@@ -225,65 +244,67 @@ public function testParseWithQos($qos, $bit)
     {
         $input =
             chr($bit) .
-            chr(4) .
+//            chr(4) .
```

Comment:
I'm currently not sure about this change. I'm sure you've don't this because it was necessary to get the tests running, but why this test was touched by your change. Can you explain this? 

Its also possible the test was wrong. But I've to have a deeper look to that.

Maybe you can also check, the original code was correct to this documentation: http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc385349263

Same with Retain and Dup tests :confused: